File: Java/Estudo-API/API-Biblioteca-DevEficiente/src/main/java/io/github/monthalcantara/biblioteca/domain/Livro.java
package io.github.monthalcantara.biblioteca.domain;

import lombok.Getter;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Entity
/*
* Adicionado indice (Além do Id) a tabela poderia ser um indice com duas
* ou mais colunas passando as colunas @Index(columnList = "isbn", "titulo")
**/
@Table(name ="Livro", indexes = @Index(columnList = "isbn"))
public class Livro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotBlank
    @Column(name = "titulo", nullable = false)
    private String titulo;

    @NotNull
    @Column(name = "preco", nullable = false)
    private BigDecimal preco;

    @NotBlank
    @Column(name = "isbn", nullable = false, unique = true)
    private String isbn;

}

File: Kotlin/API-Casa_Do_Codigo_Kotlin/src/main/kotlin/io/github/monthalcantara/casadocodigo/model/Livro.kt
package io.github.monthalcantara.casadocodigo.model

import java.math.BigDecimal
import javax.persistence.*

@Entity
data class Livro(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long? = null,
    val titulo: String,
    val resumo: String,
    val sumario: String,
    val preco: BigDecimal,
    val numeroPaginas: Int,
    val isbn: String,
    @ManyToOne val categoria: Categoria,
    @ManyToOne val autor: Autor
)